# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextSamplers.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGL2ContextSamplers.cpp
--- a/dom/canvas/WebGL2ContextSamplers.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGL2ContextSamplers.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -75,15 +75,12 @@
   sampler.SamplerParameter(pname, FloatOrInt(param));
 }
 
-void WebGL2Context::GetSamplerParameter(JSContext*, const WebGLSampler& sampler,
-                                        GLenum pname,
-                                        JS::MutableHandleValue retval) {
+MaybeWebGLVariant WebGL2Context::GetSamplerParameter(
+    const WebGLSampler& sampler, GLenum pname) {
   const FuncScope funcScope(*this, "getSamplerParameter");
-  retval.setNull();
+  if (IsContextLost()) return Nothing();
 
-  if (IsContextLost()) return;
-
-  if (!ValidateObject("sampler", sampler)) return;
+  if (!ValidateObject("sampler", sampler)) return Nothing();
 
   ////
 
@@ -97,21 +94,18 @@
     case LOCAL_GL_TEXTURE_COMPARE_FUNC: {
       GLint param = 0;
       gl->fGetSamplerParameteriv(sampler.mGLName, pname, &param);
-      retval.set(JS::Int32Value(param));
+      return AsSomeVariant(param);
     }
-      return;
-
     case LOCAL_GL_TEXTURE_MIN_LOD:
     case LOCAL_GL_TEXTURE_MAX_LOD: {
       GLfloat param = 0;
       gl->fGetSamplerParameterfv(sampler.mGLName, pname, &param);
-      retval.set(JS::Float32Value(param));
+      return AsSomeVariant(param);
     }
-      return;
 
     default:
       ErrorInvalidEnumInfo("pname", pname);
-      return;
+      return Nothing();
   }
 }
 